### PR TITLE
buildEnv: support ignoreSingleFileOutputs (resumed)

### PR DESCRIPTION
**Pull request by ShamrockLee**
Add option to ignore single-file output instead of failing, which would be useful to fix Mic92/nixpkgs-review#408.

Cc: @corngood

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Mic92 — APPROVED** [review]
Would be cool to have this because it's not possible from pure nix to know wether a derivation is a single file.

**Comment by ShamrockLee** [comment]
> Would be cool to have this because it's not possible from pure nix to know wether a derivation is a single file.

I think we should enforce a policy to exclude all single-file outputs from `meta.outputsToInstall` for all packages exposed under `pkgs` with `recurseIntoAttrs` in the long run, but it is also crucial for `buildEnv` to be flexible enough to ignore those single-file outputs.

**Comment by philiptaron** [comment]
@ShamrockLee, I'd like to merge this once the merge conflict is resolved.

**Comment by ShamrockLee** [comment]
@Mic92 @philiptaron Regarding the name, do you think it should be `ignoreFileOutputs` or `ignoreSingleFileOutputs`?

**Comment by philiptaron** [comment]
I'm amenable to both, but `ignoreSingleFileOutputs` does more clearly say what it's doing.

**Comment by ShamrockLee** [comment]
Renamed!

File: pkgs/build-support/buildenv/default.nix
Comment (philiptaron) [inline pkgs/build-support/buildenv/default.nix:26]:
```suggestion
  # Whether to ignore outputs that are a single file instead of a directory.
```

**Review by philiptaron — APPROVED** [review]
One copy-editing nit on a comment.

**Review by philiptaron — APPROVED** [review]

**Comment by ShamrockLee** [comment]
Rebasd and squashed the commit.

**Pull request merged**
